src/utils/transactionUtils/transactionUtils.ts

```diff
@@ -588,3 +623,35 @@ export function validateTTLSeconds(ttl: number) {
     );
   }
 }
+
+export function getPendingInteractionsRowsForContract(
+  pendingContractInteractions: ContractInteraction[],
+  existingValues: any,
+): {
+  attribute: string;
+  value: string;
+  id: string;
+  valid: boolean | undefined;
+}[] {
+  // find all pending interactions for the contract, find relevant ones related to row attributes
+  const pendingTxRowData = [];
+  for (const i of pendingContractInteractions) {
+    const attributes = getAttributesFromInteractionFunction(i.payload.function);
+    // TODO: this is not pretty, and could be avoided if we rework the ANT contract to allow `setTTL` and `setTransaction` rather than all of them
+    // relying only on setRecord.
+    for (const attribute of attributes) {
+      // TODO: change targetID to be transactionId
+      const attr = attribute === 'targetID' ? 'transactionId' : attribute;
```

Comment:
need something similar to the controller field 
![image](https://github.com/ar-io/pdns-react/assets/85306700/57a389e5-ac53-49a8-9d0d-9222a3c7933e)
owner as well
![image](https://github.com/ar-io/pdns-react/assets/85306700/3908b549-4bb5-40aa-9133-90446d5dd88c)

Reply:
added a util function for these and put a TODO